xapian-core/tests/api_cluster.cc

```diff
@@ -0,0 +1,40 @@
+#include <config.h>
+
+#include "api_cluster.h"
+
+#include <xapian.h>
+
+#include "apitest.h"
+#include "testutils.h"
+
+#include <string.h>
+#include <cstring>
+
+using namespace std;
+
+DEFINE_TESTCASE(euclidian_distance, backend) {
+	Xapian::Database db(get_database("etext.txt"));
+    string query = "good";
+    Xapian::Stem stemmer("en");
+
+    Xapian::QueryParser qp;
+    qp.set_database(db);
+    qp.set_stemmer(stemmer);
+
+    Xapian::Enquire enq(db);
+    enq.set_query(qp.parse_query(query));
+    Xapian::MSet matches = enq.get_mset(0, 30);
+
+	Xapian::MSetDocumentSource docs(matches);
```

Comment:
And these. If your editor is set to display tab characters as 4 columns, then you'll struggle to make this work. Tab characters should align to 8 columns, and your editor should be configured to a 4 column alignment indent.